api/base/serializers.py

```diff
@@ -1003,7 +1009,19 @@ def is_valid(self, clean_html=True, **kwargs):
         return ret
 
 
-class JSONAPISerializer(ser.Serializer):
+class PrefetchRelationshipsSerializer(ser.Serializer):
+    relationship_field_names = list()
+
+    def __init__(self, *args, **kwargs):
+        super(PrefetchRelationshipsSerializer, self).__init__(*args, **kwargs)
+        for field_name, field in self.fields.iteritems():
+            if field.source == '*':
+                self.relationship_field_names.append(field_name)
```

Comment:
Just so I understand this correctly: `relationship_field_names` is actually *all* field names? And the non-relationship fields just get ignored by `eager`? If that's the case, `relationship_field_names` is a misnomer; probably should be `field_names`. Also, `PrefetchRelationshipsSerializer` doesn't have any prefetching behavior, so we can probably get rid of it and just put this code on `JSONAPISerializer`.